Move SVG renderer logic back into SVGSkin

adroitwhiz opened the pull request:
## Depends on https://github.com/LLK/scratch-svg-renderer/pull/213

### Resolves

A step towards #550

### Proposed Changes

This PR moves the logic for actually rendering SVGs back into `SVGSkin` from `SvgRenderer`, and removes the dependency on the `SvgRenderer` class.

### Reason for Changes

This will allow for future modifications to the way SVGs are drawn, which are necessary to fix #550.

### Test Coverage

Tested manually

adroitwhiz committed:
Move SVG renderer logic back into SVGSkin

This will allow for fancier stuff to be done with the SVG viewbox in the
future to avoid subpixel jitter.

adroitwhiz commented on src/SVGSkin.js:
```diff
@@ -94,18 +103,26 @@ class SVGSkin extends Skin {
+        this._canvas.height = height * scale;
+        // Even if the canvas at the current scale has a nonzero size, the image's dimensions are floored pre-scaling.
+        // e.g. if an image has a width of 0.4 and is being rendered at 3x scale, the canvas will have a width of 1, but
+        // the image's width will be rounded down to 0 on some browsers (Firefox) prior to being drawn at that scale.
```
note to self: move this comment in the middle of the condition so it's clear it refers specifically to the naturalWidth/naturalHeight check

adroitwhiz commented on src/SVGSkin.js:
```diff
@@ -28,8 +28,17 @@ class SVGSkin extends Skin {
+        this._svgImage = null;
+
+        /** @type {Object} */
+        this._svgMeasurements = {x: 0, y: 0, width: 0, height: 0};
```
We don't need x and y right now but will need them for #550

cwillisf commented on src/SVGSkin.js:
```diff
@@ -147,7 +165,7 @@ class SVGSkin extends Skin {
         const mipScale = Math.pow(2, mipLevel - INDEX_OFFSET);
 
-        if (this._svgRenderer.loaded && !this._scaledMIPs[mipLevel]) {
+        if (this._svgImage.complete && !this._scaledMIPs[mipLevel]) {
```
It's not clear to me that this is guaranteed to never become true before the `load` event handler is called. I think to be safe we should set a flag in the `load` event handler and check that here, similar to how the `loaded` flag worked before.

adroitwhiz replied:
I've done so now. This required some additional supporting changes to avoid a race condition in case `setSVG` is called a second time before the original image is finished loading.

cwillisf reviewed (changes requested):
Thanks for doing this, and sorry it took us so long to review it!
There's one small change I've requested just to make sure it's safe but overall it looks pretty good

adroitwhiz committed:
Use one SVG image and track loaded state w/ flag

Instead of creating a new image every time `setSVG` is called and
checking whether that image is complete with `loaded`, reuse the same
image, override its `src`, & use `onload` instead of `addEventListener`
to replace the previous event listener. This is necessary to avoid race
conditions with the `_svgImageLoaded` flag.

cwillisf committed:
add comment explaining `onload` choice

cwillisf reviewed (approved):
👍
